### PR TITLE
created an entire user function so that posting comments worked

### PR DESCRIPTION
**Pull request by JamesATeasdale**

### CODE REVIEW

File: src/components/Review.jsx [hunk 1]
```diff
@@ -29,6 +34,9 @@ const Review = () => {
 
 	const handleSubmit = (e) => {
 		e.preventDefault();
+		const comment = { author: user.username, body: e.target[0].value };
+		postComment(review_id, comment);
+		console.log(review_id, comment);
```
Comment (callumu21) [inline src/components/Review.jsx]: This console.log should be removed. You also need to consider what you should do with the comment that comes back to show the user that it's been added. Currently, there's no way to know if a comment has been successfully posted.

Equally, there is no error handling here, and a failed post causes the app to crash with an unhandled promise rejection. Consider what you as a user would expect to happen in a post was unsuccessful. What kind of message would be useful?

File: src/components/Review.jsx [hunk 2]
```diff
@@ -100,12 +108,29 @@ const Review = () => {
...
-					</label>
-					<input name="comment" type="text" />
-				</form>
+				{user.username ? (
+					<form onSubmit={handleSubmit}>
+						<div className="review-card-comment">
+							<label htmlFor="comment">
+								<img src={user.avatar_url} alt="" height="65px;" />
+							</label>
+							<input name="comment" type="text" />
```
Comment (callumu21) [inline src/components/Review.jsx]: This input field isn't a controlled component at the moment. The value of this field should be handled by React and updated with an onChange handler

File: src/components/Users.jsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import { useEffect, useState } from "react";
+import { fetchUsers } from "./api";
+
+const Users = ({ user, setUser }) => {
+	const [users, setUsers] = useState([]);
+	useEffect(() => {
+		fetchUsers().then((userList) => {
```
Comment (callumu21) [inline src/components/Users.jsx]: This should have some kind of loading pattern for users with slower connections.

**Review by callumu21 — CHANGES_REQUESTED** [review]
Good start, James. The following things need adding/changing before this can be approved:

1. You have no validation on the comment form, and so it is currently possible for users to submit an empty comment.
2. Your input field for the comment body should be a controlled component, with its value being set and maintained by React rather than the DOM. See NC Notes for a look at Controlled Components.
3. Users are also able to continually press the submit button while a request is being made. You should prevent this from happening.
4. On a successful request, the form should be reset.
5. On a successful request, users should be able to see that their comment has been added. Think about when you have access to the comment object, and how you could add it to the display.
6. On an unsuccessful request, users should see a message to let them know that something went wrong. At the moment, your rejected promise is unhandled and throwing an error.
7. Your users page should have a loading pattern.

Once you've added these in, please resubmit this PR and we'll take another look 👍

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
Hey James, I have a few notes regarding some changes that I think might need to be made:

- When posting a comment the 'submit' button does not become disabled while the request is processing, this allows users on slow 3G to be able to spam the button and send several requests to the server duplicating their comment
- When posting a comment in offline mode the app crashes due to errors relating to an undefined variable
- Currently following a successful post the page re-renders and seems to re-fetch everything, it would be nice as a user to see that the comment goes into the list without seeing everything disappear from the screen

**Pull request merged**